pkg/controller/networkpolicy/external_entity.go

```diff
@@ -0,0 +1,173 @@
+		specEqual = reflect.DeepEqual(oldEEObj.Spec, curEEObj.Spec)
+	} else {
+		oldEEObj := oldEE.(*v1alpha1.ExternalEntity)
+		curEEObj := curEE.(*v1alpha1.ExternalEntity)
+		specEqual = reflect.DeepEqual(oldEEObj.Spec, curEEObj.Spec)
+	}
+	// TODO: Right now two ExternalEntities are only considered equal if the list of Endpoints and
+	//  all NamedPorts in each Endpoint are of the exact order. Considering implementing custom compare
+	//  method for the ExternalEntity spec to solve this and improve performance.
+	if labelsEqual && specEqual {
+		klog.V(4).Infof("No change in ExternalEntity %s/%s. Skipping NetworkPolicy evaluation.", curEE.GetNamespace(), curEE.GetName())
+		return
+	}
+	// Find groups matching the old ExternalEntity's labels.
+	oldAppliedToGroupKeySet := n.filterAppliedToGroupsForPodOrExternalEntity(oldEE)
+	oldAddressGroupKeySet := n.filterAddressGroupsForPodOrExternalEntity(oldEE)
+	// Find groups matching the new ExternalEntity's labels.
+	curAppliedToGroupKeySet := n.filterAppliedToGroupsForPodOrExternalEntity(curEE)
+	curAddressGroupKeySet := n.filterAddressGroupsForPodOrExternalEntity(curEE)
+	// Create set to hold the group keys to enqueue.
+	var appliedToGroupKeys sets.String
+	var addressGroupKeys sets.String
+	// AppliedToGroup keys must be enqueued only if the ExternalEntity's spec has changed or
+	// if ExternalEntity's label change causes it to match new Groups.
+	if !specEqual {
+		appliedToGroupKeys = oldAppliedToGroupKeySet.Union(curAppliedToGroupKeySet)
+	} else if !labelsEqual {
+		// No need to enqueue common AppliedToGroups as they already have latest Pod
+		// information.
+		appliedToGroupKeys = oldAppliedToGroupKeySet.Difference(curAppliedToGroupKeySet).Union(curAppliedToGroupKeySet.Difference(oldAppliedToGroupKeySet))
+	}
+	// AddressGroup keys must be enqueued only if the ExternalEntity's spec has changed or
+	// if ExternalEntity's label change causes it to match new Groups.
+	if !specEqual {
+		addressGroupKeys = oldAddressGroupKeySet.Union(curAddressGroupKeySet)
+	} else if !labelsEqual {
+		// No need to enqueue common AddressGroups as they already have latest Pod
+		// information.
+		addressGroupKeys = oldAddressGroupKeySet.Difference(curAddressGroupKeySet).Union(curAddressGroupKeySet.Difference(oldAddressGroupKeySet))
+	}
+	for group := range appliedToGroupKeys {
+		n.enqueueAppliedToGroup(group)
+	}
+	for group := range addressGroupKeys {
+		n.enqueueAddressGroup(group)
+	}
+}
+
+// deleteExternalEntity retrieves all AddressGroups and AppliedToGroups which match the ExternalEntity's
+// labels and enqueues the groups key for further processing.
+func (n *NetworkPolicyController) deleteExternalEntity(old interface{}) {
+	ee, ok := old.(metav1.Object)
+	if !ok {
+		tombstone, ok := old.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.Errorf("Error decoding object when deleting ExternalEntity, invalid type: %v", old)
+			return
+		}
+		ee, ok = tombstone.Obj.(*v1alpha2.ExternalEntity)
```

Comment:
and for v1?

Reply:
Thanks for catching this. Added UT to cover this case.